extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -239,7 +239,9 @@ export abstract class ToolBase implements ITool {
 	}
 
 	private startVersionAndStatusUpdate() {
-		this._pendingVersionAndStatusUpdate = this.updateVersionAndStatus();
+		this._pendingVersionAndStatusUpdate = this.updateVersionAndStatus().catch(error => {
```

Comment:
worst case, we will see  3 pop ups at once, doesn't seem to be a scalable solution. can we wait for them and show it all together? and also, we've being using the dialog's message area to show messages, i think we should be consistent here as well.

Reply:
Agreed - we should try to minimize the number of popups (especially when an error occurs). It reduces the overall effectiveness of errors if the user is just spammed with a whole bunch of them.

Reply:
This has already been addressed in my latest iteration. All errors are collected and combined into a single message that is shown. @Charles please see the picture I shared in email.